### PR TITLE
[TypeSpec Validation] separate error message for folder-struct

### LINKED ISSUE #27523: [TypeSpec Validation] [FolderStructure] More specific error messages
This code groups multiple rules into a single error message, which makes it difficult to determine exactly which rules is being broken:

https://github.com/Azure/azure-rest-api-specs/blob/c75671ae0a5d82d7d96fda75150478e8581408d8/eng/tools/typespec-validation/src/rules/folder-structure.ts#L63-L80

It should be changed to return a unique error message for each rule:

1. `main.tsp` or `client.tsp`
2. `tspconfig.yaml`
3. `examples` if `main.tsp`

### PR DESCRIPTION
**Pull request by ckairen**
closes #27523

### CODE REVIEW

File: eng/tools/typespec-validation/src/rules/folder-structure.ts [hunk 1]
```diff
@@ -61,22 +61,25 @@ export class FolderStructureRule implements Rule {
...
-      (await host.checkFileExists(path.join(folder, "main.tsp"))) ||
-      (await host.checkFileExists(path.join(folder, "client.tsp")));
+    let mainExists = await host.checkFileExists(path.join(folder, "main.tsp"));
+    let clientExists = await host.checkFileExists(path.join(folder, "client.tsp"));
 
-    if (await host.checkFileExists(path.join(folder, "main.tsp"))) {
-      containsMinStruct =
-        containsMinStruct && (await host.checkFileExists(path.join(folder, "examples")));
+    if (!mainExists && !clientExists) {
+      errorOutput += `Invalid folder structure: Package must contain main.tsp or client.tsp.`;
```
Comment (mikeharder) [inline eng/tools/typespec-validation/src/rules/folder-structure.ts:68]:
```suggestion
      errorOutput += `Invalid folder structure: Spec folder must contain main.tsp or client.tsp.`;
```

**Review by mikeharder — APPROVED** [review]
Approved pending error message update

File: eng/tools/typespec-validation/src/rules/folder-structure.ts [hunk 2]
```diff
@@ -61,22 +61,25 @@ export class FolderStructureRule implements Rule {
...
+      success = false;
     }
 
-    if (!packageFolder.includes("Shared")) {
-      containsMinStruct =
-        containsMinStruct && (await host.checkFileExists(path.join(folder, "tspconfig.yaml")));
+    if (mainExists && !(await host.checkFileExists(path.join(folder, "examples")))) {
+      errorOutput += `Invalid folder structure: Package with main.tsp must contain examples folder.`;
```
Comment (ckairen) [inline eng/tools/typespec-validation/src/rules/folder-structure.ts:73]:
```suggestion
      errorOutput += `Invalid folder structure: Spec folder with main.tsp must contain examples folder.`;
```

File: eng/tools/typespec-validation/src/rules/folder-structure.ts [hunk 3]
```diff
@@ -61,22 +61,25 @@ export class FolderStructureRule implements Rule {
...
+      success = false;
     }
-    if (!containsMinStruct) {
+
+    if (
+      !packageFolder.includes("Shared") &&
+      !(await host.checkFileExists(path.join(folder, "tspconfig.yaml")))
+    ) {
+      errorOutput += `Invalid folder structure: Package must contain tspconfig.yaml.`;
```
Comment (ckairen) [inline eng/tools/typespec-validation/src/rules/folder-structure.ts:81]:
```suggestion
      errorOutput += `Invalid folder structure: Spec folder must contain tspconfig.yaml.`;
```

**Pull request merged**